api/v1beta1/azurecluster_webhook_test.go

```diff
@@ -105,6 +118,26 @@ func TestAzureCluster_ValidateUpdate(t *testing.T) {
 		cluster    *AzureCluster
 		wantErr    bool
 	}{
+		{
+			name: "azurecluster with pre-existing control plane endpoint - valid spec",
```

Comment:
can we also add a test case that checks it fails and returns an error when host / port is modified? and also that it doesn't return an error if host/port is set from empty

Reply:
So, I thought I did the former with this test, line 139 wants the error and oldCluster and cluster variables have different host/port for the control plane endpoints.

I will add the second test.

Reply:
I think I got the following tests in at this point, unless I'm not interpreting what this function is doing (which is entirely possible).

 * Check for error when changing a cluster with an existing endpoint
 * Check setting an empty endpoint works without error